openXCenabler/DiagViewController.swift

```diff
@@ -35,14 +35,22 @@ class DiagViewController: UIViewController, UITextFieldDelegate {
     // set default diag response target
     vm.setDiagnosticDefaultTarget(self, action: DiagViewController.default_diag_rsp)
 
+    
   }
   
   override func didReceiveMemoryWarning() {
     super.didReceiveMemoryWarning()
     // Dispose of any resources that can be recreated.
   }
 
-  
+    override func viewDidAppear(_ animated: Bool) {
+        if(!vm.isBleConnected){
+            let alertController = UIAlertController(title: "", message:
+                "BLE is not connected to the Device", preferredStyle: UIAlertControllerStyle.alert)
```

Comment:
Same alert code is at 6 places. Keep it in a different method and call that method wherever required.